source/components/button-upload/button-upload-component.js

```diff
@@ -36,12 +37,25 @@ class ButtonUpload extends PureComponent {
     btnText: 'Upload',
     disabled: false,
     limit: false,
+    type: 'default',
+    files: [],
     loading: false,
     formatWhiteList: ['.png', '.jpg', '.jpeg', '.pdf'],
     maxFileSize: 3000000, // 3MB
     allowedDimensions: [],
   };
 
+  componentDidUpdate(prevProps) {
+    if (this.props.type === 'controlled') {
+      const { files } = this.props;
+      if (prevProps.files.length !== files.length) {
```

Comment:
aqui imagino que tenha que ser feito algo mais complexo. Pq o array pode ser manter do mesmo tamanho, mas as imagens serem diferentes. Então teríamos que testar se os dois arrays são iguais

Vi até uma discussão da melhor maneira de fazer essa comparação:
https://stackoverflow.com/questions/3115982/how-to-check-if-two-arrays-are-equal-with-javascript

Reply:
Isso pensando numa possivel edição de imagem? Havia pensado apenas de fazer a lógica de remover.

Reply:
Como hoje a remoção é item a item e não tem como remover e adicionar ao mesmo tempo, acho que tudo bem verificar apenas o length pra fazer a alteração. Eu pelo menos não consegui simular um caso onde se mantivesse o mesmo length. Mas acho que deixar uma verificação mais parruda pronta é sempre uma boa!

De qualquer forma, como a gente tá pensando apenas na remoção e não na adição vinda de fora, também não tem necessidade de chamar o `this.handleChange` aqui e nem daquela adição na linha 127 que a Debs comentou também. Também não precisa mudar a assinatura da função handleChange. Basta atualizar o estado da lista para o estado do pai, que já é o mais atualizado:

```js
componentDidUpdate(prevProps) {
    const { files } = this.props;
    if (prevProps.files.length !== files.length) {
      this.setState({ list: files });
    }
  }
```

Reply:
Eu havia pensando em incluir a chamada pro handleChange pra gente não perder as validações que tem lá dentro da função. O que acham?

Reply:
Perfeito! Mas se a gente pensar só na remoção, o arquivo já foi validado no momento que ele foi adicionado, portando, não perdemos nenhuma validação! No final, ele já foi removido do estado do componente pai e estamos apenas atualizando o filho dessa notícia!